### DISCUSSION TITLE
110/220 v range

### DISCUSSION THREAD

**Discussion by jovicassia** [vortigont/espem#50] [General]
after loading the files and restarting it displays this error
I looked in the DATA folder and there is no /config.json

[    68][E][vfs_api.cpp:105] open(): /littlefs/config.json does not exist, no permits for creation
[    80][E][vfs_api.cpp:105] open(): /littlefs/config_bkp.json does not exist, no permits for creation
[ 30215][E][vfs_api.cpp:105] open(): /littlefs/config.json does not exist, no permits for creation
[ 30224][E][vfs_api.cpp:137] rename(): /config.json does not exists

FILES IN DATA FOLDER

/css/espem.css.gz
/css/menu.jpg
/css/pure.css.gz
/css/style.css.gz
/css/style_dark.css.gz
/css/style_light.css.gz
/css/wp_dark.svg.gz
/css/wp_light.svg.gz
/favicon.ico.gz
/favicon.png
/icon-144x144.png
/index.html.gz
/js/embui.js.gz
/js/espem.js.gz
/js/espem.ui.json.gz
/js/lodash.custom.js.gz
/js/tz.json.gz
/js/ui_sys.json.gz
/manifest.json.gz

**Comment by vortigont** [comment]
Hi @jovicassia,
those files does not exist initially, but will be created once you apply any changes to the configuration settings in web UI.

**Comment by jovicassia** [comment]
Thank you for the prompt response

Finally, after 2 days of trying I managed to get it working
I really liked the system

If you want I can translate it to PT-BR (Brazilian Portuguese) and Spanish

Some questions
I live in Brazil and here our electrical grid is 110 or 220 volts,
depending on the region, and your system only displays above 190 volts, is
there a parameter to configure this?
In addition, in my house I have 2 "phases", that is, 110V + 110V. If I
connect between one "phase" and the ground I have 110V and if I connect
between the two "phases" I have 220V (for example for a shower).
So I need to use 2 PZEMs to be able to monitor the "phases" separately, is
there any parameter to configure this?




[image: image.png]

[image: image.png]

The information on the dashboard is incorrect
See that I am on a 110V power grid but the pointer is at 240

And I believe that, as here, the 220V grid never goes above 230/240 (here
the norm is a maximum of 10% of the nominal value), so I think that the
pointer going up to 270V is too much (if it reaches 270V it is better to
run and get a fire extinguisher)


[image: image.png]
[image: image.png]

The interface still has some labels in Russian

[image: image.png]
My graph is not displaying the data
and the date is showing since 1975!!!
I set NTP to Brazil and TZ to America/São Paulo

[image: image.png]
POOL is enabled, and has data stored


Thank you for your attention
Sorry for the long text but I'm trying to contribute to improving the system

Em qui., 17 de out. de 2024 às 05:30, vortigont ***@***.***>
escreveu:

**Comment by jovicassia** [comment]
[image: image.png]
[image: image.png]

I'm back
I discovered 2 things
You have to update the browser's time, otherwise it will keep displaying
those dates from 1975 etc. (page title in Russian)
And the graph only works if there is voltage AND current

In the graph, the red line is the current and voltage line?

[image: image.png]

[image: image.png]
I didn't understand this value on the graph scale


When I restart the system, the TIME SERIES COLLECTION option becomes
disabled again.

and I need to update the browser time again

Em qui., 17 de out. de 2024 às 11:48, Jorge Vila / Cassia Fernandes <
***@***.***> escreveu:

**Comment by vortigont** [comment]
@jovicassia pls, do not reply via e-mail, use GitHub page to write your posts. Otherwise all your pictures are lost. I do not see any attachments here https://github.com/vortigont/espem/discussions/50
Could you pls update and clarify your issues?


Regarding the gauge range for 110 V, yes, you can change it, but not through WebUI. It's the property hidden in js code.
You can find it [here](https://github.com/vortigont/espem/blob/f26ef57fec4b87859a382d227de47cc70bc34e7d/resources/html/js/espem.js#L130) in the `"axes"` section.
To update it in the WebUI you need to ungzip the file under `data/js/espem.js.gz`, edit there pack and upload to FS, either using FTP or uploading FS image.

To make it changing it via WebUI's option is not that trivial but I will think about it.


> And the graph only works if there is voltage AND current

this is pretty obvious, if there are 0 values then it must something wrong with either PZEM or the connection, because PZEM powers from AC it can't return 0 as voltage value, so it it considered as error and discarded.


> When I restart the system, the TIME SERIES COLLECTION option becomes disabled again.

it looks like your config changes are not saved to FS for some reason. It must be saved in `/config.json` file. Is there any messages in log that could indicate an error?

> I set NTP to Brazil and TZ to America/São Paulo

If NTP server is available and reachable you must see a message in log something like "NTP adjusted" bla-bla. Setting time from the browser is there for the environments where Internet and NTP is not accessible at all. Can you check if NTP is actually accessible from your network?


As a general concern this project is a bit outdated due to I do not have currently any PZEM devices in possession, so can't run and verify it all from my side. But let's see if I can understand the issues with your help and fix.

The EmbUI used here is also outdated, current version is much more featured but the code of this project needs to be adopted to the new API. Also it might be tricky to build it with a new version. If you are ready to do testing and provide feedbacks from your side I'll see if I can invest some time to update it.

Cheers!
